pre_processing/read_slstr_main.F90

```diff
@@ -234,42 +221,51 @@ subroutine read_slstr(infile, imager_geolocation, imager_measurements, &
 
    ! This bit reads all the data.
-   do i=1, n_bands
+   do i = 1, n_bands
       if (verbose) write(*,*) 'Reading SLSTR data for band', band_ids(i)
       if (band_ids(i) .lt. 7) then
          call read_slstr_visdata(indir, band_ids(i), &
               imager_measurements%data(:,:,i), imager_angles, startx, starty, &
-              nx, ny, nx, ny, 0, 1)
+              nx, ny)
       else if (band_ids(i) .le. 9) then
          call read_slstr_tirdata(indir, band_ids(i), &
-              imager_measurements%data(:,:,i), &
-              startx, starty, nx, ny, nx, ny, 1, 1)
+              imager_measurements%data(:,:,i), startx, starty)
       else if (band_ids(i) .lt. 16) then
-         call read_slstr_visdata(indir, band_ids(i), &
-              imager_measurements%data(:,:,i), &
-              imager_angles, startx, starty, obnx, obny, nx, ny, obl_off-1, 2)
+         if (sx_nad .gt. 0 .and. ex_nad .gt. 0) then
+            call read_slstr_visdata(indir, band_ids(i), &
+                 imager_measurements%data(sx_nad:ex_nad,:,i), &
+                 imager_angles, sx_obl, starty, ex_obl-sx_obl+1, ny)
+         end if
       else if (band_ids(i) .le. 18) then
-         call read_slstr_tirdata(indir, band_ids(i), &
-              imager_measurements%data(:,:,i), &
-              startx, starty, obnx, obny, nx, ny, obl_off, 2)
+         if (sx_nad .gt. 0 .and. ex_nad .gt. 0) then
+            call read_slstr_tirdata(indir, band_ids(i), &
+                 imager_measurements%data(sx_nad:ex_nad,:,i), sx_obl, starty)
+         end if
       else
          write(*,*) 'Invalid band_id! Must be in range 1->18', band_ids(i)
-         stop
+         stop error_stop_code
       end if
+
       ! Apply some correction factors
       if (band_ids(i) .eq. 1) then
-        imager_measurements%data(:,:,i) = imager_measurements%data(:,:,i) * 0.95
+         imager_measurements%data(:,:,i) = imager_measurements%data(:,:,i) * 0.95
       else if (band_ids(i) .eq. 5) then
-        imager_measurements%data(:,:,i) = imager_measurements%data(:,:,i) * 1.1
+         imager_measurements%data(:,:,i) = imager_measurements%data(:,:,i) * 1.1
       else if (band_ids(i) .eq. 6) then
-        imager_measurements%data(:,:,i) = imager_measurements%data(:,:,i) * 1.1
+         imager_measurements%data(:,:,i) = imager_measurements%data(:,:,i) * 1.1
       else if (band_ids(i) .eq. 10) then
-        imager_measurements%data(:,:,i) = imager_measurements%data(:,:,i) * 0.92
+         imager_measurements%data(:,:,i) = imager_measurements%data(:,:,i) * 0.92
       else if (band_ids(i) .eq. 11) then
-        imager_measurements%data(:,:,i) = imager_measurements%data(:,:,i) * 0.93
+         imager_measurements%data(:,:,i) = imager_measurements%data(:,:,i) * 0.93
       else if (band_ids(i) .eq. 12) then
-        imager_measurements%data(:,:,i) = imager_measurements%data(:,:,i) * 0.93
+         imager_measurements%data(:,:,i) = imager_measurements%data(:,:,i) * 0.93
       end if
```

Comment:
Can we have these defined somewhere obvious, like CONSTANT_OBLIQUE_OFFSET is, so that they're easy to change in the future? I worry that if, like before, they're buried here then people won't notice them.

Reply:
They can be, but I thought the idea was the code should be deleted ASAP?

Reply:
Yeah, I agree that they should be...but IMO it's better to make this obvious, just in case the L1 calibration isn't fixed within the timespan of our memory ;-)